src/allocation.rs

```diff
@@ -297,3 +297,55 @@ pub fn pod_collect_to_vec<
   dst_bytes[..src_size].copy_from_slice(src_bytes);
   dst
 }
+
+/// An extension trait for `TransparentWrapper` and alloc types.
+pub trait TransparentWrapperAlloc<Inner: ?Sized>: TransparentWrapper<Inner> {
+  /// Convert a vec of the inner type into a vec of the wrapper type.
+  fn wrap_vec(s: Vec<Inner>) -> Vec<Self>
+  where
+    Self: Sized,
+    Inner: Sized
+  {
+    // Not sure if these are needed
+    assert!(size_of::<*mut Inner>() == size_of::<*mut Self>());
+    assert!(align_of::<*mut Inner>() == align_of::<*mut Self>());
+
+    let mut s = core::mem::ManuallyDrop::new(s);
+    unsafe {
+      // SAFETY:
+      // * ptr comes from Vec (and will not be double-dropped)
+      // * the two types have the identical representation
+      // * the len and capacity fields are valid
+      Vec::from_raw_parts(
+        s.as_mut_ptr() as *mut Self,
```

Comment:
Someone reminded me that the call to .as_mut_ptr has to come *after* the calls to get the len and capacity, otherwise there's a stupid Stacked Borrows error and the pointer gets invalidated. I'd put each of these on their own line, and with as_mut_ptr as the last method to be called.